### PR TITLE
verifier.go: Add checks for request method and stateful headers

### PR DESCRIPTION
**Pull request by irori**
This implements Step 3-5 of the "Cross-origin trust" algorithm
([Section 4 of draft-yasskin-http-origin-signed-responses.html](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#rfc.section.4)).

### CODE REVIEW

File: go/signedexchange/verifier.go [hunk 1]
```diff
@@ -115,16 +116,47 @@ func (e *Exchange) Verify(verificationTime time.Time, certFetcher CertFetcher, l
...
+		}
+
+		// Step 5: "If exchange's headers contain a stateful header field, as
+		//         defined in Section 4.1, return "invalid"."
+		for k := range e.RequestHeaders {
+			if IsStatefulRequestHeader(k) {
+				l.Printf("Exchange has stateful request header %q.", k)
+				continue SignatureLoop
+			}
+		}
```
Comment (nyaxt) [inline go/signedexchange/verifier.go]: Would you factor out a helper method so we wont have label jump `continue`s?

File: go/signedexchange/verifier.go [hunk 2]
```diff
@@ -115,16 +116,47 @@ func (e *Exchange) Verify(verificationTime time.Time, certFetcher CertFetcher, l
...
 		_, err = verifySignature(e, verificationTime, certFetcher, signature)
 		if err != nil {
 			l.Printf("Verification of sinature %q failed: %v", signature.Label, err)
 			continue
 		}
-		// TODO: Implement Step 3-7.
+
+		// Step 3: "Let exchange be the exchange metadata and headers parsed out
+		//         of headers."
+		// e contains the exchange metadata and headers.
```
Comment (nyaxt) [inline go/signedexchange/verifier.go]: L132 seems broken?

File: go/signedexchange/verifier.go (hunk 1 above)
Comment (irori) [reply go/signedexchange/verifier.go]: Done.

File: go/signedexchange/verifier.go (hunk 2 above)
Comment (irori) [reply go/signedexchange/verifier.go]: Not actually, 'e' refers to a local variable.
Added `` ` `` around `e` hoping to make it easier to understand.

**Pull request merged**